### ISSUE TITLE
Ox::Builder#to_s returns trailing newline

### ISSUE THREAD

**Issue by groe** [ohler55/ox#147]
The Ox::Builder result string contains a trailing newline character.

```ruby
builder = Ox::Builder.new
builder.text("lorem ipsum")
builder.close
builder.to_s
=> "lorem ipsum\n"
```

Thats a good practice when saving the result in file (i.e. using either the `Builder::file` or the `Builder::io` class method). I guess that's why it's here in the first place.

However, when saving the result string in a database or doing further processing with it (as it is expected when `Builder::new` and `Builder#to_s` are used), this trailing newline has no use I think (correct me if i'm wrong). How about removing it in this case?

**Comment by ohler55** [comment]
It is not really possible for Ox to know that the generated string will be used for a database or some other purpose. I can add an option to not add a trailing new line character though. How does that sound?

**Comment by groe** [comment]
What I meant is that `to_s` could (unlike the IO- and file-based Builder instances) always omit the trailing newline. But an option sounds good as well!

So, together with `indent: 0`, this should ensure that there isn't any additional XML formatting inserted, right?

**Comment by ohler55** [comment]
If it is just the to_s method then it would be easier at this point to just strip! or rstrip!. Combined with ```indent: 0``` no additional white space would be added.

**Comment by groe** [comment]
That's what I do right now with the string returned from `to_s` (although I use `chomp`) and it has close to zero impact on performance (even though I do it many times for many builder instances).

So, either way,  removing the whitespace using strip/chomp, or not adding it in the first place for string-target `Builder` instances, if `to_s` could omit the whitespace I would be happy :)

**Comment by ohler55** [comment]
Please try the latest on github. Using an indent of -1 will omit the trailing newline.

**Comment by groe** [comment]
Thank you so much @ohler55!
Will try and report back.

**Comment by groe** [comment]
Works perfectly!

**Comment by ohler55** [comment]
Great, just released 2.4.4.

**Comment by ohler55** [comment]
Great, just released 2.4.4.
